postgres/datadog_checks/postgres/explain_parameterized_queries.py

```diff
@@ -72,18 +73,40 @@ def __init__(self, check, config):
     @tracked_method(agent_check_getter=agent_check_getter)
     def explain_statement(self, dbname, statement, obfuscated_statement):
         if self._check.version < V12:
-            return None
+            # if pg version < 12, skip explaining parameterized queries because
+            # plan_cache_mode is not supported
+            e = psycopg2.errors.UndefinedParameter("Unable to explain parameterized query")
+            logger.debug(
+                "Unable to explain parameterized query. Postgres version %s does not support plan_cache_mode",
+                self._check.version,
+            )
+            return None, DBExplainError.parameterized_query, '{}'.format(type(e))
         self._set_plan_cache_mode(dbname)
 
         query_signature = compute_sql_signature(obfuscated_statement)
-        if not self._create_prepared_statement(dbname, statement, obfuscated_statement, query_signature):
-            return None
+        try:
+            self._create_prepared_statement(dbname, statement, obfuscated_statement, query_signature)
+        except Exception as e:
+            # if we fail to create a prepared statement, we cannot explain the query
+            return None, DBExplainError.failed_to_explain_with_prepared_statement, '{}'.format(type(e))
 
-        result = self._explain_prepared_statement(dbname, statement, obfuscated_statement, query_signature)
-        self._deallocate_prepared_statement(dbname, query_signature)
-        if result:
-            return result[0][0][0]
-        return None
+        try:
+            result = self._explain_prepared_statement(dbname, statement, obfuscated_statement, query_signature)
+            if result:
+                plan = result[0][0][0]
+                return plan, DBExplainError.explained_with_prepared_statement, None
+            else:
+                # the explain function was executed but no plan was returned
+                logger.debug(
+                    "Unable to explain parameterized query. "
+                    "The explain function %s was executed but no plan was returned",
+                    self._config.statement_samples_config.get('explain_function', 'datadog.explain_statement'),
```

Comment:
Adding the default `'datadog.explain_statement'` seems incorrect here? Don't we want whatever the used value was?

Reply:
the used function name will either be user configured explain_function or the default `datadog.explain_statement`. See the function name we passed to the explain query here
https://github.com/DataDog/integrations-core/blob/d14e912d73f9ae4e650bece442c7ad5920b01db7/postgres/datadog_checks/postgres/explain_parameterized_queries.py#L152-L154

Reply:
Ok this is suboptimal duplication of a significant string value, but not the hill I'm going to die on.